tensorflow/core/protobuf/config.proto

```diff
@@ -99,6 +99,37 @@ message GPUOptions {
   // the overall host system performance.
   bool force_gpu_compatible = 8;
 
+  // Whether to merge data transfer streams into the compute stream in a stream
+  // group. Four types of streams will be created within a stream group by
+  // default: one compute stream, one host_to_device copy stream, one
+  // device_to_host copy stream, and several device_to_device copy streams. Use
+  // the StreamMergeOptions to merge the copy streams into the compute stream.
+  // For example, setting "merge_host_to_device_stream = true" to make the
+  // compute stream responsible for both computation and host to device memory
+  // copy. Stream merging helps reduce the overhead caused by stream
+  // synchronization, especially when data transfers are frequent.
+  message StreamMergeOptions {
+    // If true, the compute stream will be used for host_to_device copy as well.
+    // It's no longer necessary to record an event before the copy to let the
+    // copy stream wait for the compute stream to finish. There is also no need
+    // to wait for the copy to complete before executing the callback function.
+    bool merge_host_to_device_stream = 1;
+
+    // If true, the compute stream will be used for device_to_host copy as well.
+    // It's no longer necessary to record an event before the copy to let the
+    // copy stream wait for the compute stream to finish.
+    bool merge_device_to_host_stream = 2;
+
+    // If true, the compute stream will be used for device_to_device copy as
+    // well. It's no longer necessary to record an event before the copy to let
+    // the copy stream wait for the compute stream of the sending device to
+    // finish. There is also no need to wait for the compute stream of the
+    // receiving device to finish if the copy is within the same device.
+    bool merge_device_to_device_stream = 3;
+  }
+
+  StreamMergeOptions stream_merge_options = 10;
```

Comment:
Can you move this option to the inside of Experimental? Non-experimental API changes will need a wider agreement. For now, let's start with experimental options.